denokv/main.rs

```diff
@@ -53,19 +77,146 @@ struct AppState {
 
 #[tokio::main]
 async fn main() -> Result<(), anyhow::Error> {
-  let config = Config::parse();
-  std::env::set_var("RUST_LOG", "info");
+  let config: &'static Config = Box::leak(Box::new(Config::parse()));
+  if std::env::var("RUST_LOG").is_err() {
+    std::env::set_var("RUST_LOG", "info");
+  }
   env_logger::init();
 
-  if config.access_token.len() < 12 {
+  match &config.subcommand {
+    SubCmd::Serve(options) => {
+      if options.read_only && options.sync_from_s3 {
+        anyhow::bail!("Cannot sync from S3 in read-only mode.");
```

Comment:
Clap can already enforce this, see above.

Can you add a check that none of the `options.replica` are specified when `sync_from_s3` is unset?

Reply:
> Can you add a check that none of the options.replica are specified when sync_from_s3 is unset?

The S3 options are also taken from env vars, so checking this would break if the user has dirty environment variables.